src/set_theory/game/pgame.lean

```diff
@@ -1065,4 +1069,72 @@ begin
   exact zero_lt_one,
 end
 
+/-- The birthday of a pre-game is inductively defined as the least strict upper bound of the
+birthdays of its left and right games. It may be thought as the "step" in which a certain game is
+constructed. -/
+noncomputable def birthday : pgame.{u} → ordinal.{u}
+| ⟨xl, xr, xL, xR⟩ :=
+    max (ordinal.lsub.{u u} $ λ i, birthday (xL i)) (ordinal.lsub.{u u} $ λ i, birthday (xR i))
+
+@[simp] theorem birthday_mk (xl xr xL xR) : birthday (mk xl xr xL xR) = max
```

Comment:
Arguably you don't need this lemma, you can just put `simp` on the `def` to have the same effect.

Note your proofs already mix `rw birthday` and `rw birthday_mk` anyway; they mean the same thing.